src/main.go

```diff
@@ -473,41 +570,41 @@ func getDmwActiveContactStateData(apiUrl string, tenants [1]string) (models.DmwK
 	if err != nil {
 		fmt.Print(err.Error())
 		os.Exit(1)
-		return dmwKnownContacts, err
+		return dmwKnownContact, err
 	}
 
 	if response.StatusCode == 200 {
 		responseData, err = ioutil.ReadAll(response.Body)
 		if err != nil {
 			fmt.Print(err.Error())
-			return dmwKnownContacts, err
+			return dmwKnownContact, err
 		}
 	} else {
 		fmt.Printf(method+httpBadResponse, response.StatusCode, response.Status)
 	}
 
 	if responseData != nil {
-		err = json.Unmarshal(responseData, &dmwKnownContacts)
+		err = json.Unmarshal(responseData, &dmwKnownContact)
 		if err != nil {
 			fmt.Printf("[%s] cannot unmarshal dmw response: [%v]\n", method, err)
-			return dmwKnownContacts, err
+			return dmwKnownContact, err
 		}
 	}
 
-	sort.SliceStable(dmwKnownContacts.Contacts, func(i, j int) bool {
-		return dmwKnownContacts.Contacts[i].ContactID < dmwKnownContacts.Contacts[j].ContactID
+	sort.SliceStable(dmwKnownContact.Contacts, func(i, j int) bool {
+		return dmwKnownContact.Contacts[i].ContactID < dmwKnownContact.Contacts[j].ContactID
 	})
 
-	return dmwKnownContacts, err
+	return dmwKnownContact, err
 }
 
 // getDfoAuthToken calls the DFO api POST engager/2.0/token to get a bearer token for subsequent DFO api calls
-func getDfoAuthToken(apiUrl string, clientId string, clientSecret string) (models.DfoAuthTokenObj, error) {
+func getDfoAuthToken(apiUrl string, i InputDataObj) (models.DfoAuthTokenObj, error) {
 	contentType := "application/json"
 	var dfoAuthTokenBody models.DfoAuthTokenBody
 	dfoAuthTokenBody.GrantType = "client_credentials"
-	dfoAuthTokenBody.ClientId = clientId
-	dfoAuthTokenBody.ClientSecret = clientSecret
+	dfoAuthTokenBody.ClientId = i.ClientId
+	dfoAuthTokenBody.ClientSecret = i.ClientSecret
```

Comment:
If these are required, do you still want to make the API call or is the plan for the call to the API to error and that will say that "clientid and secret are missing" as the output?

Reply:
The way this program is working currently, we ask for this info at the very beginning in the console.  I did this for lack of knowing where we were going to run this so I am just feeding in the data we need for now.  If you don't have this, then you can't even begin to run the program.